diff -ru roundcubemail/config/defaults.inc.php roundcubemail-cookielifetime/config/defaults.inc.php
--- roundcubemail/config/defaults.inc.php	2023-07-01 13:29:36.000000000 +0200
+++ roundcubemail-cookielifetime/config/defaults.inc.php	2023-07-06 13:30:38.189429928 +0200
@@ -586,9 +586,13 @@
 // 0 - hide product name and version number, 1 - show product name only, 2 - show product name and version number
 $config['display_product_info'] = 1;
 
-// Session lifetime in minutes
+// Session lifetime in minutes (server side)
 $config['session_lifetime'] = 10;
 
+// Cookie lifetime in minutes for session id and auth cookies
+// 0 - session cookies (expire on browser exit)
+$config['cookie_lifetime'] = 0;
+
 // Session domain: .example.org
 $config['session_domain'] = '';
 
diff -ru roundcubemail/program/lib/Roundcube/rcube.php roundcubemail-cookielifetime/program/lib/Roundcube/rcube.php
--- roundcubemail/program/lib/Roundcube/rcube.php	2023-07-01 13:29:36.000000000 +0200
+++ roundcubemail-cookielifetime/program/lib/Roundcube/rcube.php	2023-07-06 13:30:38.189429928 +0200
@@ -456,6 +456,7 @@
         $sess_path     = $this->config->get('session_path');
         $sess_samesite = $this->config->get('session_samesite');
         $lifetime      = $this->config->get('session_lifetime', 0) * 60;
+        $c_lifetime    = $this->config->get('cookie_lifetime', 0) * 60;
         $is_secure     = $this->config->get('use_https') || rcube_utils::https_check();
 
         // set session domain
@@ -475,8 +476,7 @@
             ini_set('session.gc_maxlifetime', $lifetime * 2);
         }
 
-        // set session cookie lifetime so it never expires (#5961)
-        ini_set('session.cookie_lifetime', 0);
+        ini_set('session.cookie_lifetime', $c_lifetime);
         ini_set('session.cookie_secure', $is_secure);
         ini_set('session.name', $sess_name ?: 'roundcube_sessid');
         ini_set('session.use_cookies', 1);
diff -ru roundcubemail/program/lib/Roundcube/rcube_session.php roundcubemail-cookielifetime/program/lib/Roundcube/rcube_session.php
--- roundcubemail/program/lib/Roundcube/rcube_session.php	2023-07-01 13:29:36.000000000 +0200
+++ roundcubemail-cookielifetime/program/lib/Roundcube/rcube_session.php	2023-07-06 13:30:38.193429921 +0200
@@ -713,8 +713,11 @@
      */
     public function set_auth_cookie()
     {
+        $clifetime = $this->config->get('cookie_lifetime', 0) * 60;
+        if (!$clifetime) { $expiration = 0; }
+        else { $expiration = time() + $clifetime; }
         $this->cookie = $this->_mkcookie($this->now);
-        rcube_utils::setcookie($this->cookiename, $this->cookie, 0);
+        rcube_utils::setcookie($this->cookiename, $this->cookie, $expiration);
         $_COOKIE[$this->cookiename] = $this->cookie;
     }
 
